account_test.go

```diff
@@ -0,0 +1,218 @@
+package dcrlibwallet_test
+
+import (
+	"os"
+	"strconv"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/raedahgroup/dcrlibwallet"
+)
+
+const rootDir = ".dcrlibwallet_tests_root"
+
+var _ = Describe("Account", func() {
+	var (
+		wallet *Wallet
+	)
+
+	BeforeEach(func() {
+		multi, err := NewMultiWallet(rootDir, "", "testnet3")
+		Expect(err).To(BeNil())
+		Expect(multi).ToNot(BeNil())
+
+		wallet, err = multi.CreateNewWallet("0000", 0)
+		Expect(err).To(BeNil())
+		Expect(wallet).ToNot(BeNil())
+	})
+
+	AfterEach(func() {
+		if wallet != nil {
+			wallet.Shutdown()
+		}
+		err := os.RemoveAll(rootDir)
+		Expect(err).To(BeNil())
+	})
+
+	Describe("GetAccounts(requiredConfirmations int32) (string, error)", func() {
+		Context("when called", func() {
+			It("should get accounts", func() {
+
+				res, err := wallet.GetAccounts(0)
+				By("returning a nil error")
+				Expect(err).To(BeNil())
+				By("returning nonempty accounts string")
+				Expect(res).ToNot(BeEquivalentTo(""))
+			})
+		})
+	})
+
+	Describe("GetAccount(accountNumber int32, requiredConfirmations int32) (*Account, error)", func() {
+		Context("when called with the right parameters", func() {
+			It("should get account", func() {
+
+				var accountNumber int32 = 0
+				account, err := wallet.GetAccount(accountNumber, 0)
+				By("returning a nil error")
+				Expect(err).To(BeNil())
+				By("returning a non nil account")
+				Expect(account).ToNot(BeNil())
+				By("returning an account with the supplied account number")
+				Expect(account.Number).To(BeEquivalentTo(accountNumber))
+			})
+		})
+
+		Context("when called for non-existing account", func() {
+			It("it should fail", func() {
+
+				account, err := wallet.GetAccount(1000000000, 0)
+				By("returning a non nil error")
+				Expect(err).ToNot(BeNil())
+				By("by returning nil account")
+				Expect(account).To(BeNil())
+			})
+		})
+	})
+
+	Describe("GetAccountBalance(accountNumber int32, requiredConfirmations int32) (*Balance, error)", func() {
+		Context("when called", func() {
+			It("should get the account balance", func() {
+
+				balance, err := wallet.GetAccountBalance(0, 0)
+				By("returning a nil error")
+				Expect(err).To(BeNil())
+				By("returning the expected balance")
+				Expect(balance).ToNot(BeNil())
```

Comment:
Just not `nil` is not the expected balance. If the wallet was just created, then the balance should be equivalent to 0.